core/client/fs/src/main/java/alluxio/client/block/stream/DataWriter.java

```diff
@@ -62,13 +62,15 @@ public static DataWriter create(FileSystemContext context, long blockId, long bl
           alluxioConf.getBoolean(PropertyKey.USER_SHORT_CIRCUIT_PREFERRED);
       boolean ufsFallbackEnabled = options.getWriteType() == WriteType.ASYNC_THROUGH
           && alluxioConf.getBoolean(PropertyKey.USER_FILE_UFS_TIER_ENABLED);
+      boolean workerIsLocal = CommonUtils.isLocalHost(address, alluxioConf);
 
-      if (context.hasProcessLocalWorker() && !ufsFallbackEnabled) {
+      if (workerIsLocal && context.hasProcessLocalWorker() && !ufsFallbackEnabled) {
+        LOG.debug("Creating worker process local output stream for block {} @ {}",
+            blockId, address);
         return BlockWorkerDataWriter.create(context, blockId, blockSize, options);
       }
 
-      if (CommonUtils.isLocalHost(address, alluxioConf)
-          && shortCircuit
+      if (workerIsLocal && shortCircuit
```

Comment:
Other than the logs when the if conditions are all met, I think it is also very important to know which condition is NOT met. In other words, we may expect the if condition to be true but when it is not true, there should be some hints how it is not true. I'd recommend this:
```
if (conditionA && conditionB && conditionC) {
  LOG.debug(...)
  // do sth
  // Actually if you can return here, just return explicitly to avoid confusing
}
LOG.debug("a message that contains conditionA={} conditionB={} conditionC={}")
...
```
This is not logging more than you are already doing right now, because we are logging one line when the IF block executes, we should also accept logging one message for the implicit ELSE block.